Git Formats/Git Blame Ignore Revisions.sublime-syntax

```diff
@@ -0,0 +1,51 @@
+%YAML 1.2
+---
+# Syntax based on documentation here:
+# https://git-scm.com/docs/git-blame#Documentation/git-blame.txt---ignore-revs-fileltfilegt
+# (refers to https://git-scm.com/docs/git-fsck#Documentation/git-fsck.txt-fsckskipList)
+# https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view
+
+name: Git Blame Ignore Revisions
+scope: text.git.blame.ignorerevs
+version: 2
+
+file_extensions:
+  - git-blame-ignore-revs     # *.git-blame-ignore-revs
+
+hidden_file_extensions:
+  - .git-blame-ignore-revs    # ~/.git-blame-ignore-revs
+
+variables:
+  comment_char: '[#]'         # Can't use Git Common here - ';' is not allowed
+  object_name: \b\h{40}\b
+
+contexts:
+  prototype:
+    - include: comments
+
+  main:
+    - include: revision
+    - include: illegal
+
+  comments:
+    - match: ({{comment_char}}).*
```

Comment:
We want to scope the `\n` in comments, if present as some plugins rely on it. Maybe uncommon for this file, but generally applied to all syntaxes.

```suggestion
    - match: ({{comment_char}}).*\n?
```

Reply:
Generally it's better to match the comment start token, then push into a named context for the comment contents, which will pop after consuming the `\n` char. Why?
Because it helps with both embedding and extending.

Reply:
Agree. Wouldn't have argued for it though as this pattern was copied from _Git Common.sublime-syntax_, most patterns in _Git Commit_ etc. also use blump capture groups only and - am I wrong? - those files seem rather uncommon to be candidates for extending them with interpolation.

Changing that would be welcome, however.

Reply:
Something like this?

```yaml
  comment-start:
    - match: '[{{comment_char}}]'
      scope: punctuation.definition.comment.git.blame.ignorerevs
      push: comment-content

  comment-content:
    - meta_scope: comment.line.git.blame.ignorerevs
    - match: .*\n?
      pop: 1
```

Reply:
```yaml
comment-start:
  - match: '{{comment_char}}'
    scope: punctuation....
    push: ...

comment-content:
  - meta_scope: comment...
  - match: $\n?
    pop: 1
```

Pardon any typos or shortening. I'm on a phone.

Reply:
The point is explicitly to avoid the `.*` in the match.

Reply:
IIRC the `$` is relevant only on EOF, but at that point the pop is not really important. 😉

Reply:
I have made changes. Question: wouldn't set be better than push here for overwriting the `rest` context if applicable?

Reply:
No, because `set` would replace `main` with `rest` on stack.

Reply:
Nitpicking, but ST has guards that don't allow `main` to be replaced and instead push a new context on the stack, right? (That wouldn't apply if this syntax were to be extended, of course.)

Reply:
It has some safe guards, but still setting `rest` onto stack is wrong - nothing else is of interest. It would cause trouble as soon as this syntax is included in another one via `- include: scope:text.git...`